### DISCUSSION TITLE
[Run issue]: Problem in specifying different regions in the world for using different physics modules

### DISCUSSION THREAD

**Discussion by SFR1993** [topas-nbio/TOPAS-nBio#36] [Run-Problem] [answered]
### What was the issue?

Hello to topas-nbio code experts. I have been learning and using this code recently. In the **_"MultiScale_AuNP_TOPAS-main"_** example, two different regions are defined to use two physical models; _tareg_ and _aunpreg_. Although a same method has been used to define these two areas, but after running the code, I get the following error:

 G4Exception : GeomMgt1001
 issued by : G4RegionStore::GetRegion()
Region NOT found in store!
 Region aunpreg NOT found in store!
 Returning NULL pointer.

*I should mention that this code written in TOPAS v3.2, but I try to run it in v4.0 (OpenTOPAS).

*Here is part of the code:

# ----- Geometry -----
# World
d:Ge/World/HLX       = 30 um
d:Ge/World/HLY       = 30 um
d:Ge/World/HLZ       = 30 um
s:Ge/World/Material  = "G4_WATER"
b:Ge/World/Invisible = "True"
s:Ge/World/Color     = "Aqua"
s:Ge/World/DrawingStyle = "WireFrame"

# Target
s:Ge/Target/Type 			= "G4Orb"
s:Ge/Target/Parent 			= "World"
s:Ge/Target/Material 		        = "G4_WATER"
d:Ge/Target/R 				= 10.1 um
s:Ge/Target/Color 			= "aqua"
b:Ge/Target/Invisible 		        = "True"

# Cell
 s:Ge/TheCell/Type 			= "G4Orb"
 s:Ge/TheCell/Parent 			= "Target"
 s:Ge/TheCell/Material 		        = "G4_WATER"
 d:Ge/TheCell/R 			= 10 um
 s:Ge/TheCell/Color 			= "green"
 b:Ge/TheCell/Invisible 		= "True"

# Nucleus
d:Ge/TheCell/Nucleus/FibreLength   = 120 nm  # please set <= 120 nm
d:Ge/TheCell/Nucleus/NucleusRadius = 6.9 um  # Oelze et al (4T1)
b:Ge/TheCell/Nucleus/AddHydrationShell     = "true"   # "true" or "false"
s:Ge/TheCell/Nucleus/FileName              = "HilbertPoints_iteration1.dat"
i:Ge/TheCell/Nucleus/HilbertCurveLayer     = 2
i:Ge/TheCell/Nucleus/HilbertCurve3DRepeat  = 36
d:Ge/TheCell/Nucleus/ToleranceSpace  = .14913333278374 um
d:Ge/TheCell/Nucleus/NuclTransX = 0. um
d:Ge/TheCell/Nucleus/NuclTransY = 0. um
d:Ge/TheCell/Nucleus/NuclTransZ = 0. um

#AuNPs
i:Ge/TheCell/AuNPs/NbOfAuNPsInCytoplasm = 1000000 #35610 # must be defined to have AuNPs (may be 0)
i:Ge/TheCell/AuNPs/NbOfAuNPsInNucleus = 0
d:Ge/TheCell/AuNPs/MaxDistFromNucleus = 100 nm
d:Ge/TheCell/AuNPs/Rmax = 1 nm
d:Ge/TheCell/AuNPs/Rcore = 1 nm
s:Ge/TheCell/AuNPs/Material = "G4_Au"
s:Ge/TheCell/AuNPC/Material = "G4_Au"
s:Ge/TheCell/AuNPN/Material = "G4_Au"
s:Ge/TheCell/AuNPCore/Material = "G4_Fe" # Gets replaced in extension
s:Ge/TheCell/AuNPC/Color = "orange"
s:Ge/TheCell/AuNPC/DrawingStyle = "solid"
s:Ge/TheCell/AuNPN/Color = "aqua"
s:Ge/TheCell/AuNPN/DrawingStyle = "solid"

# ----- Physics -----
Ge/CheckForOverlaps = "False"

sv:Ph/Default/Modules = 2 "g4em-livermore" "g4em-dna-chemistry" # "g4em-dna_opt2" # "TsEmDNAPhysics" # "g4em-dna-chemistry" # 
b:Ph/Default/Fluorescence = "True"
b:Ph/Default/Auger = "True"
b:Ph/Default/PIXE = "True"
b:Ph/Default/AugerCascade = "True"
b:Ph/Default/DeexcitationIgnoreCut = "True"

includeFile = ./supportFiles/TOPASChemistry.txt
s:Ch/ChemistryName = "TOPAS"
b:Ch/TOPAS/ChemicalStageTransportActive = "true" # set transportaion flag
d:Ch/TOPAS/ChemicalStageTimeEnd = 1.0 ns  #set the duration for chemical satge
dv:Ch/TOPAS/AddTimeStepHighEdge = 1 999999 ps
dv:Ch/TOPAS/AddTimeStepResolution = 1 0.5 ps

# G4DNA settings
s:Ge/Target/AssignToRegionNamed = "tareg"
s:Ph/Default/ForRegion/tareg/ActiveG4EmModelFromModule = "g4em-dna-chemistry" # "g4em-dna" # "TsEmDNAPhysics" # 
d:Ph/Default/ForRegion/tareg/SetProductionCutLowerEdge = 15 eV
d:Ph/Default/ForRegion/tareg/SetProductionCutHighEdge  = 1 MeV
d:Ph/Default/ForRegion/tareg/EMRangeMin = 15 eV
d:Ph/Default/ForRegion/tareg/EMRangeMax = 1 MeV
d:Ph/Default/ForRegion/tareg/CutForElectrons = 0.1 nm

#Assign region
s:Ge/TheCell/AuNPs/AssignToRegionNamed = "aunpreg"
s:Ge/TheCell/AuNPN/AssignToRegionNamed = "aunpreg"
s:Ge/TheCell/AuNPC/AssignToRegionNamed = "aunpreg"

# Livermore settings
s:Ph/Default/ForRegion/aunpreg/ActiveG4EmModelNamed = "g4em-livermore" 

### TOPAS-nBio category

Example

### Version

3.0

### What Operative System (OS) are you running on?

Linux (Debian/Ubuntu)

### Relevant log output

_-------- WWWW ------- G4Exception-START -------- WWWW -------
*** G4Exception : GeomMgt1001
      issued by : G4RegionStore::GetRegion()
Region NOT found in store !
        Region aunpreg NOT found in store !
        Returning NULL pointer.
_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Attestation

- [X] I testify that I have fully read the bug reporting guideline beforehand.

**Comment by JoseRamosMendez** [comment, marked as answer]
Hi there!
You have not defined the (sub)component mentioned in `Ge/TheCell/AuNPs/` so the region `aunpreg` cannot be created/attached. Is that AuNPs created with a geometry extension created by you? Also, in TOPAS-nBio v3.0 with TOPAS vOpenTOPAS4.0 you can use gold cross sections, see the example `$TOPAS-nBio/examples/processes/GoldNanoParticle.txt`
Cheers.

**Reply by SFR1993** [reply]
Dr. Ramos Mendez,
Thank you for responding so quickly. No, it was not made by me. I got it from "_github.com/AKlapproth/MultiScale_AuNP_TOPAS_" and I just tried to run it to see how I can use two physics models in different regions, because I need to use such this definition in my research. 

Best regards,
Sedigheh.
